keras_cv/models/stable_diffusion/ptp_utils.py

```diff
@@ -0,0 +1,262 @@
+        tokens_edit (np.ndarray): Array of the edit prompt tokens.
+
+    Returns:
+        mask (np.ndarray): Mask of the original and edited prompt overlap.
+        indices: (np.ndarray): Indices of the original and edited prompt overlap.
+    """
+    tokens_conditional = tokenizer.encode(prompt)
+    tokens_conditional_edit = tokenizer.encode(prompt_edit)
+    mask, indices = seq_aligner.get_mapper(tokens_conditional, tokens_conditional_edit)
+    return mask, indices
+
+
+def set_initial_tf_variables(diff_model: tf.keras.Model):
+    """Create initial control variables to auxiliate the prompt editing method.
+
+    Args:
+        diff_model (tf.keras.Model): The diffusion model.
+    """
+    for submodule in diff_model.submodules:
+        submodule_name = submodule.name
+        if "cross_attention" in submodule_name:
+            # Set control variables
+            submodule.cross_attn_mode = tf.Variable(
+                "", dtype=tf.string, trainable=False
+            )
+            submodule.use_prompt_weights = tf.Variable(
+                False, dtype=tf.bool, trainable=False
+            )
+            # Set array variables
+            submodule.attn_map = tf.Variable(
+                [], shape=tf.TensorShape(None), dtype=tf.float32, trainable=False
+            )
+            submodule.prompt_edit_mask = tf.Variable(
+                [], shape=tf.TensorShape(None), dtype=tf.float32, trainable=False
+            )
+            submodule.prompt_edit_indices = tf.Variable(
+                [], shape=tf.TensorShape(None), dtype=tf.int32, trainable=False
+            )
+            submodule.prompt_weights = tf.Variable(
+                [], shape=tf.TensorShape(None), dtype=tf.float32, trainable=False
+            )
+
+
+def reset_initial_tf_variables(diff_model: tf.keras.Model):
+    """Reset the control variables to their default values.
+
+    Args:
+        diff_model (tf.keras.Model): The diffusion model.
+    """
+    for submodule in diff_model.submodules:
+        submodule_name = submodule.name
+        if "cross_attention" in submodule_name:
+            # Reset control variables
+            submodule.cross_attn_mode.assign("")
+            submodule.use_prompt_weights.assign(False)
+            # Reset array variables
+            submodule.attn_map.assign([])
+            submodule.prompt_edit_mask.assign([])
+            submodule.prompt_edit_indices.assign([])
+            submodule.prompt_weights.assign([])
+
+
+def overwrite_forward_call(diff_model: tf.keras.Model):
+    """Update the attention forward pass with a custom call method.
+
+    Args:
+        diff_model (tf.keras.Model): The diffusion model.
+    """
+    for submodule in diff_model.submodules:
+        submodule_name = submodule.name
+        if "cross_attention" in submodule_name:
+            # Overwrite forward pass method
+            submodule.call = call_attn_edit.__get__(submodule)
+
+
+def call_attn_edit(self, inputs):
```

Comment:
Since this method has `self` as an input and doesn't explicitly show here how it's added to a class where "self" makes sense, can you please mention this in the docstring?

Reply:
Is it more explicit now? Should I add something more? @ianstenbit